ForumProject/communications/consumers.py

```diff
@@ -0,0 +1,110 @@
+import json
+
+from asgiref.sync import async_to_sync
+from channels.generic.websocket import WebsocketConsumer
+
+from .models import Room, Message
+
+
+class ChatConsumer(WebsocketConsumer):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(args, kwargs)
+        self.room_name = None
+        self.room_group_name = None
+        self.room = None
+        self.messages = None
+        self.user = None
+        self.user_inbox = None
+
+    def connect(self):
+        self.room_name = self.scope['url_route']['kwargs']['room_name']
+        self.room_group_name = f'chat_{self.room_name}'
+        self.room = Room.objects.get(name=self.room_name)
+        self.messages = Message.objects.filter(room=self.room)
+        self.user = self.scope['user']
+        self.user_inbox = f'inbox_{self.user.first_name}'
+
+        self.accept()
+
+        async_to_sync(self.channel_layer.group_add)(
+            self.room_group_name,
+            self.channel_name,
+        )
+
+        self.send(json.dumps({
+            'type': 'user_list',
+            'users': [user.first_name for user in self.room.online.all()],
+        }))
+
+        if self.user.is_authenticated:
```

Comment:
You check `if self.user.is_authenticated` only after sending some initial data to the client. It's better to check user authentication at the very beginning of the `connect()` method to ensure only authenticated users can connect.

Reply:
Done, thanks